### PR TITLE
Feat/1207977371934597 vesting krest solution

### PR DESCRIPTION
**Pull request by sfffaaa**
Note: 
1. in the current vesting info (1028), it's OK to traverse all the vesting schedules directly. I have already tested that on the local environment.
2. We have to make sure we only did this upgrade once. Otherwise, the schedule will be wrong.

### CODE REVIEW

File: runtime/krest/src/vesting_migration.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			weight_writes
+		);
+		T::DbWeight::get().reads_writes(weight_reads, weight_writes)
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+		let mut old_schedules = Vec::new();
+		for (_acc_id, mut schedules) in pallet_vesting::Vesting::<T>::iter() {
+			if schedules.len() == 0 {
```
Comment (DocteurPing) [inline runtime/krest/src/vesting_migration.rs]: Can't we do something like this instead:
```
if schedules.len() != 0 {
 			old_schedules = schedules.drain(..).collect();
			break;
}
```

File: runtime/krest/src/vesting_migration.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), TryRuntimeError> {
+		let old_schedules =
+			<Vec<VestingInfo<BalanceOf<T>, BlockNumberFor<T>>> as Decode>::decode(&mut &state[..])
+				.expect("pre_upgrade_step provides a valid state; qed");
+
+		let mut new_schedules = Vec::new();
+		for (_acc_id, mut schedules) in pallet_vesting::Vesting::<T>::iter() {
+			if schedules.len() == 0 {
```
Comment (DocteurPing) [inline runtime/krest/src/vesting_migration.rs]: Can't we do something like this instead:
```
if schedules.len() != 0 {
 			old_schedules = schedules.drain(..).collect();
			break;
}
```

File: runtime/krest/src/vesting_migration.rs (hunk 1 above)
Comment (sfffaaa) [reply runtime/krest/src/vesting_migration.rs]: Refined

File: runtime/krest/src/vesting_migration.rs (hunk 2 above)
Comment (sfffaaa) [reply runtime/krest/src/vesting_migration.rs]: Refined

**Review by DocteurPing — APPROVED** [review]

**Pull request merged**